bitcoin/src/blockdata/witness.rs

```diff
@@ -368,9 +372,9 @@ impl Witness {
     }
 
     fn element_at(&self, index: usize) -> Option<&[u8]> {
-        let varint = VarInt::consensus_decode(&mut &self.content[index..]).ok()?;
-        let start = index + varint.size();
-        Some(&self.content[start..start + varint.0 as usize])
+        let (compact, size) = compact_size::decode(&self.content[index..]).ok()?;
```

Comment:
Also expect.

Reply:
We don't know that `index` is valid here because it comes indirectly from the argument passed in to `nth`. I don't believe we want to panic here if it is invalid.

Reply:
It's a private method that's always called with a valid `index`. We might as well document that as precondition. But since it's used at only one place we might as well move its code there and remove the method.